Python/remote_debugging.c

```diff
@@ -422,26 +422,127 @@ search_map_for_section(proc_handle_t *handle, const char* secname, const char* m
+        CloseHandle(hFile);
+        return NULL;
+    }
+
+    BYTE* mapView = (BYTE*)MapViewOfFile(hMap, FILE_MAP_READ, 0, 0, 0);
+    if (!mapView) {
+        PyErr_SetFromWindowsErr(0);
+        CloseHandle(hMap);
+        CloseHandle(hFile);
+        return NULL;
+    }
+
+    IMAGE_DOS_HEADER* pDOSHeader = (IMAGE_DOS_HEADER*)mapView;
+    if (pDOSHeader->e_magic != IMAGE_DOS_SIGNATURE) {
+        PyErr_SetString(PyExc_RuntimeError, "Invalid DOS signature.");
+        UnmapViewOfFile(mapView);
+        CloseHandle(hMap);
+        CloseHandle(hFile);
+        return NULL;
+    }
+
+    IMAGE_NT_HEADERS* pNTHeaders = (IMAGE_NT_HEADERS*)(mapView + pDOSHeader->e_lfanew);
+    if (pNTHeaders->Signature != IMAGE_NT_SIGNATURE) {
+        PyErr_SetString(PyExc_RuntimeError, "Invalid NT signature.");
+        UnmapViewOfFile(mapView);
+        CloseHandle(hMap);
+        CloseHandle(hFile);
+        return NULL;
+    }
+
+    IMAGE_SECTION_HEADER* pSection_header = (IMAGE_SECTION_HEADER*)(mapView + pDOSHeader->e_lfanew + sizeof(IMAGE_NT_HEADERS));
+    void* runtime_addr = NULL;
+
+    for (int i = 0; i < pNTHeaders->FileHeader.NumberOfSections; i++) {
+        const char* name = (const char*)pSection_header[i].Name;
+        if (strncmp(name, substr, IMAGE_SIZEOF_SHORT_NAME) == 0) {
+            runtime_addr = (void*)(remote_base + pSection_header[i].VirtualAddress);
+            UnmapViewOfFile(mapView);
+            CloseHandle(hMap);
+            CloseHandle(hFile);
+            return runtime_addr;
+        }
+    }
+
+    UnmapViewOfFile(mapView);
+    CloseHandle(hMap);
+    CloseHandle(hFile);
+
+    return NULL;
+}
+
+
 static uintptr_t
 search_map_for_section(proc_handle_t* handle, const char* secname, const char* substr) {
-    //TODO: Implement this function
-    PyErr_SetString(PyExc_RuntimeError, "search_map_for_section not implemented on Windows");
-    return 0;
+    HANDLE hProcSnap = CreateToolhelp32Snapshot(TH32CS_SNAPMODULE, handle->pid);
+    if (hProcSnap == INVALID_HANDLE_VALUE) {
+        PyErr_SetString(PyExc_PermissionError, "Unable to create module snapshot. Check permissions or PID.");
+        return 0;
+    }
+
+    MODULEENTRY32W moduleEntry;
+    moduleEntry.dwSize = sizeof(moduleEntry);
+    void* runtime_addr = NULL;
+
+    if (Module32FirstW(hProcSnap, &moduleEntry)) {
+        do {
+            char szModule[MAX_MODULE_NAME32 + 1];
+            char szExePath[MAX_PATH];
+
+            wcstombs(szModule, moduleEntry.szModule, sizeof(szModule) - 1);
+            wcstombs(szExePath, moduleEntry.szExePath, sizeof(szExePath) - 1);
+
+            // Look for either python executable or DLL
+            if (strstr(szModule, substr) || strstr(szExePath, substr)) {
+                runtime_addr = _analyze_pe(moduleEntry.szExePath, moduleEntry.modBaseAddr, secname);
+                if (runtime_addr != NULL) {
+                    CloseHandle(hProcSnap);
+                    return (uintptr_t)runtime_addr;
```

Comment:
It would be slightly cleaner to just `break` here, instead, to avoid duplicating the resource relinquishment and returning that happens at the end of the function.
```suggestion
                    break;
```